src/g_canvas.c

```diff
@@ -2071,6 +2071,32 @@ t_pd *glob_evalfile(t_pd *ignore, t_symbol *name, t_symbol *dir)
     return x;
 }
 
+t_pd* glob_evaltext(const char* data, size_t length, t_symbol* name, t_symbol* dir)
```

Comment:
that looks wrong.
i think the proper interface would be to pass a *binbuf* to be evaluated, not a raw text.
Pd already provides functions to parse a raw text into a binbuf.

Reply:
Pd also provides a function to load a file into a binbuf, so why do we need `glob_evalfile`? The idea is to have a nice high level function to load a patch from a given text buffer. The signature of `glob_evaltext` is certainly not "wrong". I can see how `glob_evalbinbuf` could be more flexible, so why not have both? The former could simply delegate to the latter.

Reply:
on the other hand, we could have `glob_evalbinbuf` in Pd and `libpd_opentext` in libpd (complementing `libpd_openfile`).

Reply:
> The signature of `glob_evaltext` is certainly not "wrong"

sorry; i didn't want to imply that the signature looks technically wrong.

a `glob_evaltext` (or `libpd_opentext`) funtion would look like:

~~~
t_binbuf*b=binbuf_new();
binbuf_text(b, data, length);
glob_evalbinbuf(b, name, dir);
binbuf_free(b);
~~~

i don't see much merit in such a simple wrapper function consisting of four (4) calls to public functions.

Reply:
yes, the wrapper function could go to libpd.